lib/mix/lib/mix/tasks/format.ex

```diff
@@ -112,6 +156,71 @@ defmodule Mix.Tasks.Format do
     end
   end
 
+  # This function reads exported configuration from the imported dependencies and deals with
+  # caching the result of reading such configuration in a manifest file.
+  defp fetch_deps_opts(formatter_opts) do
+    deps = formatter_opts[:import_deps]
+
+    cond do
+      is_nil(deps) or deps == [] ->
+        formatter_opts
+
+      is_list(deps) ->
+        dep_parenless_calls =
+          if deps_dot_formatters_stale?() do
+            dep_parenless_calls = eval_deps_opts(deps)
+
+            if dep_parenless_calls != [] do
+              write_deps_manifest(dep_parenless_calls)
```

Comment:
If we don't write the file, the manifest is always out of date and we will execute this every time. It is probably best to write the file even if it will be an empty list.

Reply:
That makes sense, but do we want to write it even if folks don't provide `:export_locals_without_parens`? That might be why I wasn't using `Keyword.get(..., [])`.

Reply:
This is the case where the user specifies some deps but they don't export anything.